### ISSUE TITLE
Open Type opens generated Java Code in Xtend Editor

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#333]
Observed this in a eclipse with oxygen sr2 as base.
have no idea what happens

**Comment by cdietrich** [comment]
![bildschirmfoto_2017-12-20_22-08-39](https://user-images.githubusercontent.com/238736/34228819-5ef5ce5e-e5d2-11e7-9db2-11f1c1a90fb2.png)

**Comment by cdietrich** [comment]
<img width="1103" alt="bildschirmfoto 2017-12-20 um 22 11 19" src="https://user-images.githubusercontent.com/238736/34228925-bd423ea2-e5d2-11e7-9d3e-47690b6be6db.png">

as you can see happens on different machines with different OS

**Comment by cdietrich** [comment]
@szarnekow do we some magic here?

**Comment by cdietrich** [comment]
looks like org.eclipse.xtext.ui.editor.XtextEditor.doSetInput(IEditorInput) is called directly with leaving subclass out

**Comment by cdietrich** [comment]
```
        <editor
              class="org.eclipse.xtend.ide.XtendExecutableExtensionFactory:org.eclipse.xtext.ui.editor.XtextEditor"
              contributorClass="org.eclipse.ui.editors.text.TextEditorActionContributor"
              default="true"
              extensions="xtend"
              icon="icons/xtend_file.png"
              id="org.eclipse.xtend.core.Xtend"
              matchingStrategy="org.eclipse.xtend.ide.XtendExecutableExtensionFactory:org.eclipse.xtext.xbase.ui.editor.JavaEditorInputMatcher"
              name="Xtend Editor"
              symbolicFontName="org.eclipse.xtend.ui.editors.textfont">
        </editor>
```

**Comment by cdietrich** [comment]
```
	public Class<? extends XbaseEditor> bindXbaseEditor() {
		return XtendEditor.class;
	}
```

why is this not working?

**Comment by cdietrich** [comment]
ahhh xbase editor is bad
@kthoms looks like there went something wrong with the refactoring

**Comment by FStolte** [comment]
I could reproduce the issues and fixed them with your first suggestion, i.e. changing the editor binding from `bindXbaseEditor()` to `bindXtextEditor()`. Since this problem was introduced in [eclipse/xtext-eclipse#463] (which happens to still be open), I will provide a pull request referencing that issue.

**Comment by FStolte** [comment]
Ok, I am sure that changing the binding manually had fixed the problem for me, but now after regenerating all xbase languages, the open type menu still opens the java file...
I will check if I can again reach the state where the correct file is opened and then try to figure out why the regeneration does not fix it.

**Comment by cdietrich** [comment]
have a look at xbaseeditor.dosetinput

**Comment by FStolte** [comment]
So it turns out that if the .xtend file is already open in an editor tab, the open type menu switches to it correctly. That is probably what made me think I already fixed it. Then I guess I'll have to keep looking...

**Comment by FStolte** [comment]
Nevermind my previous two comments, after debugging on Xtext 2.13 for a while (to figure out what should actually be happening), I switched back to the current state and its working. Well, almost.
When the .java file is not open, the open type menu opens the .xtend file, as it should be. Only when the .java file is open already the editor switches to it instead of the .xtend file. This was probably the case when I tested it after regenerating the languages, which I didn't realize.

This issue, however, is not related to [eclipse/xtext-eclipse#463] since it also happens on 2.13.

Here is a description of the older bug that remains after my fix:
If the .java file is already open and the .xtend file is not, the .java file is opened by the open type menu. If both files are open, then the open type menu seems to select the one that was most recently focussed. In any case, the file is opened in an xtend editor even if it is the .java file.

tl;dr: the new bug is fixed, an older bug remains

**Comment by cdietrich** [comment]
can you create a separate ticket for the "old" bug

**Comment by FStolte** [comment]
Under which project should I create the issue? I already checked the domainmodel example and the generated java files from that language are also affected by the bug, which means that it is not specific to xtend but rather xbase languages. So should it be under xtext-extras?

**Comment by kthoms** [comment]
Yes, xtext-extras

**Comment by cdietrich** [comment]
No xtext-eclipse

**Comment by kthoms** [comment]
Closing; original issue is fixed. New issue https://github.com/eclipse/xtext/issues/2419 was created.
